service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/gcp/ainotebook/post-startup.sh

```diff
@@ -143,98 +175,170 @@ trap 'exit_handler $? $LINENO $BASH_COMMAND' EXIT
 # Let the UI know the script has started
 set_guest_attributes "${STATUS_ATTRIBUTE}" "STARTED"
 
+echo "Determining JupyterLab environment (jupyter.service or docker)"
+
+readonly INSTANCE_CONTAINER="$(get_metadata_value instance/attributes/container)"
+
+if [[ -n "${INSTANCE_CONTAINER}" ]]; then
+  echo "Custom container detected: ${INSTANCE_CONTAINER}."
+else
+  echo "Non-containerized Jupyter experienced detected."
+fi
+
 echo "Resynchronizing apt package index..."
 
 # The apt package index may not be clean when we run; resynchronize
 apt-get update
 
+# Create the target directories for installing into the HOME directory
+${RUN_AS_JUPYTER_USER} "mkdir -p '${USER_BASH_COMPLETION_DIR}'"
+${RUN_AS_JUPYTER_USER} "mkdir -p '${USER_HOME_LOCAL_BIN}'"
+${RUN_AS_JUPYTER_USER} "mkdir -p '${USER_HOME_LOCAL_SHARE}'"
+
+# Indicate the start of Terra customizations of the ~/.bash_profile
+cat << EOF >> "${USER_BASH_PROFILE}"
+
+### BEGIN: Terra-specific customizations ###
+
+# Prepend "${USER_HOME_LOCAL_BIN}" (if not already in the path)
+if [[ ":\${PATH}:" != *":${USER_HOME_LOCAL_BIN}:"* ]]; then 
+  export PATH="${USER_HOME_LOCAL_BIN}":"\${PATH}"
+fi
+EOF
+
+# Update the PATH for container JupyterLab
+if [[ -n "${INSTANCE_CONTAINER}" ]]; then
+
+cat << EOF >> "${CONTAINER_NOTEBOOK_CONFIG}"
+
+### BEGIN: Terra-specific customizations ###
+
+import os
+
+os.environ['PATH'] = "${USER_HOME_LOCAL_BIN}:" + os.environ['PATH']
+EOF
+
+fi
+
 echo "Installing common packages via pip..."
 
 # Install common packages. Use pip instead of conda because conda is slow.
-/opt/conda/bin/pip install \
+${RUN_AS_JUPYTER_USER} "pip install --user \
   dsub \
   nbdime \
   nbstripout \
   pandas_gbq \
   pre-commit \
   pylint \
-  pytest
+  pytest"
```

Comment:
any chance we could install plotnine here, too?  [PF-2798](https://broadworkbench.atlassian.net/jira/software/c/projects/PF/issues/PF-2798)